client/src/components/PopoverOptionSet/PopoverOptionSetToggle.js

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Button } from 'reactstrap';
+import PopoverOptionSet from './PopoverOptionSet';
+
+/**
+ * Simple component that presents a button that will toggle a PopoverOptionSet.
+ * Using this component means that you won't be able to manage "open" state yourself.
+ */
+class PopoverOptionSetToggle extends Component {
+  constructor(props) {
+    super(props);
+
+    this.handleToggle = this.handleToggle.bind(this);
+
+    this.state = {
+      isOpen: false,
+    };
+  }
+
+  /**
+   * Handle toggling the "open" state of the popover.
+   */
+  handleToggle() {
+    this.setState({
+      isOpen: !this.state.isOpen,
+    });
+  }
+
+  render() {
+    const { isOpen } = this.state;
+    const { id, buttonProps: forwardedButtonProps, ...forwardedProps } = this.props;
+
+    const popoverProps = {
+      ...forwardedProps,
+      toggle: this.handleToggle,
+      isOpen,
+      target: id,
+    };
+
+    const buttonProps = {
+      ...forwardedButtonProps,
+      id,
+      onClick: this.handleToggle,
+    };
+
+    return (
+      <div>
+        <Button {...buttonProps} >
+          Toggle
+        </Button>
+        <PopoverOptionSet {...popoverProps} />
```

Comment:
First I find it a little bit strange that the toggle component is also rendering the option set itself. It could just be a case of updating the component's doc block to clarify, at the moment it sounded to me like it should just be a toggle button of some sort.

Second, the component would ideally be injected.

Reply:
This was only created because there was a lot of boilerplate I was writing in the story for this component - mostly around tracking the state of `isOpen`. That's why this is rendering the component itself. I will refactor to make the `PopoverOptionSet` injected and I'll update any documentation to indicate it's a simple example - using the `PopoverOptionSet` component on its own may be more fit for purpose.

Reply:
I guess it also makes sense that the word "Toggle" is configurable and the default is translated 🤦‍♂️

Reply:
Done 🙂